### PR TITLE
Refactor encoding

### LINKED ISSUE #555: ensure that the naming of the encodings are pretty
### Description of feature

e.g. one-hot instead of one_hot_encoding etc

### PR DESCRIPTION
**Pull request by Zethson**
Fixes #555 #576

I will adapt the tutorials quickly before this but want feedback first.

### CODE REVIEW

File: ehrapy/anndata/anndata_ext.py [hunk 1]
```diff
@@ -86,11 +86,10 @@ def df_to_anndata(
             ) from e
     else:
         obs = pd.DataFrame(index=df.index.map(str))
-        logg.info("Added all columns to `obs`.")
     dataframes = BaseDataframes(obs, df)
     numerical_columns = list(dataframes.df.select_dtypes("number").columns)
     # if data is numerical only, short-circuit AnnData creation to have float dtype instead of object
-    all_num = True if len(numerical_columns) == len(list(dataframes.df.columns)) else False
```
Comment (Imipenem) [inline ehrapy/anndata/anndata_ext.py:93]: Why does this got removed?

File: ehrapy/anndata/anndata_ext.py [hunk 2]
```diff
@@ -106,7 +105,6 @@ def df_to_anndata(
         X=X,
         obs=_cast_obs_columns(dataframes.obs),
         var=pd.DataFrame(index=list(dataframes.df.columns)),
-        dtype="float32" if all_num else "object",
```
Comment (Imipenem) [inline ehrapy/anndata/anndata_ext.py:109]: Does this work well with all numeric anndata objects? Or is this patched in latest anndata version?

File: ehrapy/preprocessing/_encode.py [hunk 3]
```diff
@@ -72,11 +72,21 @@ def encode(
         >>> # encode col1 and col2 using label encoding and encode col3 using one hot encoding
         >>> adata_encoded = ep.pp.encode(adata,
         >>>                              autodetect=False,
-        >>>                              encodings={'label_encoding': ['col1', 'col2'], 'one_hot_encoding': ['col3']})
+        >>>                              encodings={'label': ['col1', 'col2'], 'one-hot': ['col3']})
     """
     if isinstance(data, AnnData):
         # basic type checking for passed parameters when encoding a single AnnData object
-        _check_anndata_input_type(autodetect, encodings)
+        if not isinstance(autodetect, bool):
```
Comment (Imipenem) [inline ehrapy/preprocessing/_encode.py:79]: Type hint for autodetect is bool or dict; what is this check for?

File: ehrapy/anndata/anndata_ext.py (hunk 2 above)
Comment (Zethson) [reply ehrapy/anndata/anndata_ext.py:109]: I now have 
```

    all_num = True if len(numerical_columns) == len(list(dataframes.df.columns)) else False
    X = X.astype(np.number) if all_num else X.astype(object)
    # cast non numerical obs only columns to category or bool dtype, which is needed for writing to .h5ad files
    adata = AnnData(
        X=X,
        obs=_cast_obs_columns(dataframes.obs),
        var=pd.DataFrame(index=list(dataframes.df.columns)),
        layers={"original": X.copy()},
        uns=uns,
    )
```

which should solve both of your comments

File: ehrapy/preprocessing/_encode.py (hunk 3 above)
Comment (Zethson) [reply ehrapy/preprocessing/_encode.py:79]: Removed.

File: tests/anndata/test_anndata_ext.py [hunk 4]
```diff
@@ -429,7 +430,7 @@ def test_assert_numeric_vars(self):
     def test_set_numeric_vars(self):
         """Test for the numeric vars setter."""
         values = np.array(
-            [[1.2, 2.2], [3.2, 4.2], [5.2, 6.2]],
+            [[1.0, 2.2], [1.0, 4.2], [0, 6.2]],
```
[the remark this answers is not in the document]
Comment (Zethson) [reply tests/anndata/test_anndata_ext.py:433]: Ah nvm, I needed to specify a label encoding somewhere else

**Pull request merged**